events/source.go

```diff
@@ -94,6 +94,12 @@ func (es *Source) KeyChord(rn rune, code key.Codes, mods key.Modifiers) {
 // MouseButton creates and sends a mouse button event with given values
 func (es *Source) MouseButton(typ Types, but Buttons, where image.Point, mods key.Modifiers) {
 	ev := NewMouse(typ, but, where, mods)
+	if typ == MouseUp && es.Last.MouseButtonType == MouseDown {
```

Comment:
sorry to be a PITA but the key thing is to just *exclude* MouseDown -- MouseDrag definitely needs these things.

Reply:
or maybe that goes in a different place?  anyway, just exclude MouseDown to be clearer.

Reply:
You know what the great thing about MouseDrag is? It goes through MouseMove, not MouseButton. The only things that go through MouseButton are MouseDown and MouseUp, but if you really want me to invert the logic for you, I can.

Reply:
Good?